crypto/rsa/rsa_ossl.c

```diff
@@ -476,13 +483,80 @@ static int rsa_ossl_private_decrypt(int flen, const unsigned char *from,
         if (!rsa_blinding_invert(blinding, ret, unblind, ctx))
             goto err;
 
+    /*
+     * derive the Key Derivation Key from private exponent and public
+     * ciphertext
+     */
+    if (!(rsa->flags & RSA_FLAG_EXT_PKEY)) {
+        d_for_hash = BN_CTX_get(ctx);
+        if (d_for_hash == NULL) {
+            ERR_raise(ERR_LIB_RSA, ERR_R_MALLOC_FAILURE);
+            goto err;
+        }
+        if (rsa->d == NULL) {
+            ERR_raise(ERR_LIB_RSA, RSA_R_MISSING_PRIVATE_KEY);
+            goto err;
+        }
+        BN_with_flags(d_for_hash, rsa->d, BN_FLG_CONSTTIME);
+        j = BN_bn2binpad(d_for_hash, buf, num);
+        if (j < 0) {
+            ERR_raise(ERR_LIB_RSA, ERR_R_INTERNAL_ERROR);
+            goto err;
+        }
+
+        md = EVP_MD_fetch(rsa->libctx, "sha256", NULL);
+        if (md == NULL) {
+            ERR_raise(ERR_LIB_RSA, ERR_R_INTERNAL_ERROR);
+            goto err;
+        }
+
+        if (EVP_Digest(buf, num, d_hash, NULL, md, NULL) != 1) {
```

Comment:
<= 0

Reply:
it's documented as returning 1 for success, not a positive number in general

Reply:
Yeah, but we do not check failures of other functions returning 1 for success and 0 for failure with func() != 1 either. And the function is documented to return failure with 0 and nothing else.

Please change this to <= 0 as that is the most usual pattern across the sources for EVP_Digest/DigestInit.

Reply:
ok